### PR TITLE
DBZ-3901 Guard against chunk signal coming when none expected

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.redhat.com/browse/DBZ-3901

### CODE REVIEW

**Comment by gunnarmorling** [comment]
@jpechane, can you explain the situation a bit?

**Comment by jpechane** [comment]
@gunnarmorling It can happen after connector restart the last window signals can be redelivered. The code was ready to handle out of order window signals but was not ready to handle situation when no signal was expected at all.

File: debezium-core/src/main/java/io/debezium/pipeline/source/snapshot/incremental/AbstractIncrementalSnapshotContext.java [hunk 1]
```diff
@@ -78,7 +78,7 @@ public AbstractIncrementalSnapshotContext(boolean useCatalogBeforeSchema) {
     }
 
     public boolean openWindow(String id) {
-        if (!id.startsWith(currentChunkId)) {
+        if (notExpectedChunk(id)) {
             LOGGER.info("Arrived request to open window with id = '{}', expected = '{}', request ignored", id, currentChunkId);
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/pipeline/source/snapshot/incremental/AbstractIncrementalSnapshotContext.java:82]:
```suggestion
            LOGGER.info("Received request to open window with id = '{}', expected = '{}', request ignored", id, currentChunkId);
```

File: debezium-core/src/main/java/io/debezium/pipeline/source/snapshot/incremental/AbstractIncrementalSnapshotContext.java [hunk 2]
```diff
@@ -88,7 +88,7 @@ public boolean openWindow(String id) {
     }
 
     public boolean closeWindow(String id) {
-        if (!id.startsWith(currentChunkId)) {
+        if (notExpectedChunk(id)) {
             LOGGER.info("Arrived request to close window with id = '{}', expected = '{}', request ignored", id, currentChunkId);
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/pipeline/source/snapshot/incremental/AbstractIncrementalSnapshotContext.java:92]:
```suggestion
            LOGGER.info("Received request to close window with id = '{}', expected = '{}', request ignored", id, currentChunkId);
```

**Review by gunnarmorling — APPROVED** [review]
@jpechane, ok, gotcha. Could you add that explanation to the code, e.g. to the new method. I think it's worth describing the reasoning for future reference. Besides that, I've made a small (unrelated) suggestion. Can you squash and merge everything once you've addressed those all? Thx!

**Pull request merged**